Respect global config.cudaSupport

SomeoneSerge opened the pull request:
###### Description of changes

Make nixpkgs respect `config.cudaSupport` to reduce divergence in outPaths caused by out-of-tree overlays.

Also proposing a new practice: require derivations to consume `config` and write `cudaSupport ? config.cudaSupport`, instead of checking `config` outside in `all-packages.nix` or such.

Rationale: we had quite a few packages not respecting `config.cudaSupport`, resulting in people (including me) adopting overlays that "fix" them. Overlays mean diverging hashes, which means we can't share cache. The easiest way to check if there are packages that don't respect `config.cudaSupport` is to grep: ```❯ ag 'enableCuda \? false'```. Checking `config` in `all-packages.nix` means grep produces false-positives

TBD: add links to example overlays, but some are [SomeoneSerge/nixpkgs-unfree](https://github.com/SomeoneSerge/nixpkgs-unfree/blob/b8685f9d290118ee7b3dece0008706637f0c14e3/nix/overlays.nix#L29), and there are more in NixOS-QChem, and `numtide/nixpkgs-unfree`

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->


CC @NixOS/cuda-maintainers

SomeoneSerge committed:
tree-wide: incorporate common out-of-tree cudaSupport overlays

SomeoneSerge committed:
tree-wide: 'enableCuda ? false' -> 'config.cudaSupport or false' to respect global defaults

SomeoneSerge committed:
tree-wide: rm `cudaSupport ? false` formal parameters

'cudaSupport ? false' -> 'cudaSupport ? config.cudaSupport or false' to respect global defaults

Packages expressions that take `cudaSupport ? false` are likely to
ignore `config.cudaSupport`. Instead, we want them to make `cudaSupport`
a required argument, or to explicitly refer to `config`

SomeoneSerge committed:
doc: update #cuda to reflect the recommended config.cudaSupport style

SomeoneSerge commented:
This commit is particularly dangerous: https://github.com/NixOS/nixpkgs/pull/224068/commits/f8a5692a5af39c6f661fce95b4e2500e28cba4ab. Ideally we'd have tests that check that when we set `config.cudaSupport = true` the respective packages are actually built with cuda.

Potential footgun sources: many packages supply multiple variants using a pattern like `pkgWithCuda = pkgVanilla.override { cudaSupport = true; }`

samuela commented:
Woohoo! Yeah there's really no excuse for packages not to be using the `config.cudaSupport` and `config.cudaCapabilities` standard at this point

SomeoneSerge commented:
@samuela @ConnorBaker here goes the more controversial one: should we make a `config.mklSupport`, and default to `config.cudaSupport -> config.mklSupport`?

SomeoneSerge commented:
> Maybe let's save that for a future PR? Is there not a situation in which someone might want cudaSupport but not mklSupport?

Then they'd explicitly set `config.mklSupport = false`.
I got to thinking in this direction, because MKL is another source of outPath deviations that we have. NixOS-QCHem and such do override it IIRC. I just build both flavours

SomeoneSerge commented:
Now, this time I don't understand what ofborg is saying. How do I reproduce this error?

SomeoneSerge commented:
Result of `nixpkgs-review pr 224068 --extra-nixpkgs-config '{ cudaCapabilities = [ "8.6" ]; }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

Artturin commented:
misclick

SomeoneSerge commented:
Result of `nixpkgs-review pr 224068 --extra-nixpkgs-config '{ cudaCapabilities = [ "8.6" ]; cudaSupport = true; }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>188 packages failed to build:</summary>
  <ul>
    <li>apacheHttpdPackages.mod_tile</li>
    <li>cloudcompare</li>
    <li>cntk</li>
    <li>dl-poly-classic-mpi</li>
    <li>elpa</li>
    <li>entwine</li>
    <li>ezminc</li>
    <li>febio</li>
    <li>gdal (python310Packages.gdal)</li>
    <li>getdp</li>
    <li>gmt</li>
    <li>gplates</li>
    <li>grass</li>
    <li>gromacsDouble</li>
    <li>gromacsDoubleMpi</li>
    <li>libLAS</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>mysql-workbench</li>
    <li>octavePackages.communications</li>
    <li>octavePackages.control</li>
    <li>octavePackages.fem-fenics</li>
    <li>octavePackages.image-acquisition</li>
    <li>octavePackages.level-set</li>
    <li>octavePackages.ltfat</li>
    <li>octavePackages.mapping</li>
    <li>octavePackages.parallel</li>
    <li>octavePackages.signal</li>
    <li>octavePackages.sparsersb</li>
    <li>octavePackages.stk</li>
    <li>octavePackages.tisean</li>
    <li>octavePackages.vibes</li>
    <li>openmvs</li>
    <li>openorienteering-mapper</li>
    <li>paraview</li>
    <li>pdal</li>
    <li>perl534Packages.Tirex</li>
    <li>perl534Packages.Tirex.devdoc</li>
    <li>perl536Packages.Tirex</li>
    <li>perl536Packages.Tirex.devdoc</li>
    <li>petsc</li>
    <li>postgresql11JitPackages.postgis</li>
    <li>postgresql11JitPackages.postgis.doc</li>
    <li>postgresql11Packages.postgis</li>
    <li>postgresql11Packages.postgis.doc</li>
    <li>postgresql12JitPackages.postgis</li>
    <li>postgresql12JitPackages.postgis.doc</li>
    <li>postgresql12Packages.postgis</li>
    <li>postgresql12Packages.postgis.doc</li>
    <li>postgresql13JitPackages.postgis</li>
    <li>postgresql13JitPackages.postgis.doc</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql13Packages.postgis.doc</li>
    <li>postgresqlJitPackages.postgis (postgresql14JitPackages.postgis)</li>
    <li>postgresqlJitPackages.postgis.doc (postgresql14JitPackages.postgis.doc)</li>
    <li>postgresql14Packages.postgis</li>
    <li>postgresql14Packages.postgis.doc</li>
    <li>postgresql15JitPackages.postgis</li>
    <li>postgresql15JitPackages.postgis.doc</li>
    <li>postgresql15Packages.postgis</li>
    <li>postgresql15Packages.postgis.doc</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.blackjax</li>
    <li>python310Packages.blackjax.dist</li>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.cartopy</li>
    <li>python310Packages.cartopy.dist</li>
    <li>python310Packages.cntk</li>
    <li>python310Packages.cntk.dist</li>
    <li>python310Packages.cvxpy</li>
    <li>python310Packages.cvxpy.dist</li>
    <li>python310Packages.datashader</li>
    <li>python310Packages.datashader.dist</li>
    <li>python310Packages.fenics</li>
    <li>python310Packages.fenics.dist</li>
    <li>python310Packages.fiona</li>
    <li>python310Packages.fiona.dist</li>
    <li>python310Packages.folium</li>
    <li>python310Packages.folium.dist</li>
    <li>python310Packages.geopandas</li>
    <li>python310Packages.geopandas.dist</li>
    <li>python310Packages.jaxopt</li>
    <li>python310Packages.jaxopt.dist</li>
    <li>python310Packages.opensfm</li>
    <li>python310Packages.opensfm.dist</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.osmnx.dist</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.plotnine.dist</li>
    <li>python310Packages.pygmt</li>
    <li>python310Packages.pygmt.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pyprecice</li>
    <li>python310Packages.pyprecice.dist</li>
    <li>python310Packages.pyslurm</li>
    <li>python310Packages.pyslurm.dist</li>
    <li>python310Packages.python-mapnik</li>
    <li>python310Packages.python-mapnik.dist</li>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-aer</li>
    <li>python310Packages.qiskit-aer.dist</li>
    <li>python310Packages.qiskit-finance</li>
    <li>python310Packages.qiskit-finance.dist</li>
    <li>python310Packages.qiskit-ibmq-provider</li>
    <li>python310Packages.qiskit-ibmq-provider.dist</li>
    <li>python310Packages.qiskit-ignis</li>
    <li>python310Packages.qiskit-ignis.dist</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.qiskit-nature.dist</li>
    <li>python310Packages.qiskit-optimization</li>
    <li>python310Packages.qiskit-optimization.dist</li>
    <li>python310Packages.qiskit.dist</li>
    <li>python310Packages.qutip</li>
    <li>python310Packages.qutip.dist</li>
    <li>python310Packages.rasterio</li>
    <li>python310Packages.rasterio.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python310Packages.wktutils</li>
    <li>python310Packages.wktutils.dist</li>
    <li>python310Packages.worldengine</li>
    <li>python310Packages.worldengine.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.cntk</li>
    <li>python311Packages.cntk.dist</li>
    <li>python311Packages.cvxpy</li>
    <li>python311Packages.cvxpy.dist</li>
    <li>python311Packages.fenics</li>
    <li>python311Packages.fenics.dist</li>
    <li>python311Packages.fipy</li>
    <li>python311Packages.fipy.dist</li>
    <li>python311Packages.hoomd-blue</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.opensfm</li>
    <li>python311Packages.opensfm.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.pygmt</li>
    <li>python311Packages.pygmt.dist</li>
    <li>python311Packages.pyprecice</li>
    <li>python311Packages.pyprecice.dist</li>
    <li>python311Packages.pyslurm</li>
    <li>python311Packages.pyslurm.dist</li>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
    <li>python311Packages.qiskit-aer</li>
    <li>python311Packages.qiskit-aer.dist</li>
    <li>python311Packages.qiskit-finance</li>
    <li>python311Packages.qiskit-finance.dist</li>
    <li>python311Packages.qiskit-ibmq-provider</li>
    <li>python311Packages.qiskit-ibmq-provider.dist</li>
    <li>python311Packages.qiskit-ignis</li>
    <li>python311Packages.qiskit-ignis.dist</li>
    <li>python311Packages.qiskit-nature</li>
    <li>python311Packages.qiskit-nature.dist</li>
    <li>python311Packages.qiskit-optimization</li>
    <li>python311Packages.qiskit-optimization.dist</li>
    <li>python311Packages.qutip</li>
    <li>python311Packages.qutip.dist</li>
    <li>python311Packages.worldengine</li>
    <li>python311Packages.worldengine.dist</li>
    <li>pytrainer</li>
    <li>pytrainer.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>qmapshack</li>
    <li>redpanda-server</li>
    <li>rtabmap</li>
    <li>saga</li>
    <li>suitesparse_4_4</li>
    <li>sumo</li>
    <li>sunshine</li>
    <li>t-rex</li>
    <li>truecrack</li>
    <li>udig</li>
    <li>xyce</li>
    <li>xyce.doc</li>
  </ul>
</details>
<details>
  <summary>284 packages built:</summary>
  <ul>
    <li>EBTKS</li>
    <li>arrayfire</li>
    <li>bicgl</li>
    <li>bicpl</li>
    <li>cdo</li>
    <li>ceres-solver</li>
    <li>ceres-solver.dev</li>
    <li>cholmod-extra</li>
    <li>colmap (colmapWithCuda)</li>
    <li>conglomerate</li>
    <li>cp2k</li>
    <li>eccodes</li>
    <li>elmerfem</li>
    <li>fftwMpi</li>
    <li>fftwMpi.dev</li>
    <li>fftwMpi.info</li>
    <li>fftwMpi.man</li>
    <li>forge</li>
    <li>freecad</li>
    <li>g2o</li>
    <li>g2o.debug</li>
    <li>gama</li>
    <li>gegl</li>
    <li>gegl.dev</li>
    <li>gegl.devdoc</li>
    <li>gimp</li>
    <li>gimp-with-plugins</li>
    <li>gimp.dev</li>
    <li>gimpPlugins.bimp</li>
    <li>gimpPlugins.exposureBlend</li>
    <li>gimpPlugins.farbfeld</li>
    <li>gimpPlugins.fourier</li>
    <li>gimpPlugins.gap</li>
    <li>gimpPlugins.gimplensfun</li>
    <li>gimpPlugins.gmic</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.lqrPlugin</li>
    <li>gimpPlugins.resynthesizer</li>
    <li>gimpPlugins.texturize</li>
    <li>gimpPlugins.waveletSharpen</li>
    <li>globalarrays</li>
    <li>gnome-photos</li>
    <li>gnome-photos.installedTests</li>
    <li>gnudatalanguage</li>
    <li>gromacs</li>
    <li>gromacsMpi (gromacsCudaMpi)</li>
    <li>hal-hardware-analyzer</li>
    <li>hdf5-mpi</li>
    <li>hdf5-mpi.dev</li>
    <li>highfive-mpi</li>
    <li>hp2p</li>
    <li>hpcg</li>
    <li>hpl</li>
    <li>hpx</li>
    <li>hwloc</li>
    <li>hwloc.dev</li>
    <li>hwloc.doc</li>
    <li>hwloc.lib</li>
    <li>hwloc.man</li>
    <li>inormalize</li>
    <li>ior</li>
    <li>katago</li>
    <li>lammps-mpi</li>
    <li>libminc</li>
    <li>librsb</li>
    <li>libvdwxc</li>
    <li>migrate</li>
    <li>minc_tools</li>
    <li>minc_widgets</li>
    <li>mni_autoreg</li>
    <li>mpi</li>
    <li>mpich</li>
    <li>mprime</li>
    <li>mvapich</li>
    <li>n3</li>
    <li>naev</li>
    <li>nco</li>
    <li>ncview</li>
    <li>nest-mpi</li>
    <li>netcdf</li>
    <li>netcdf-mpi</li>
    <li>netcdfcxx4</li>
    <li>netcdffortran</li>
    <li>neuron-full (python310Packages.neuronpy)</li>
    <li>neuron-mpi</li>
    <li>nwchem</li>
    <li>octave</li>
    <li>octaveFull</li>
    <li>octavePackages.arduino</li>
    <li>octavePackages.audio</li>
    <li>octavePackages.bim</li>
    <li>octavePackages.bsltl</li>
    <li>octavePackages.cgi</li>
    <li>octavePackages.data-smoothing</li>
    <li>octavePackages.database</li>
    <li>octavePackages.dataframe</li>
    <li>octavePackages.dicom</li>
    <li>octavePackages.divand</li>
    <li>octavePackages.doctest</li>
    <li>octavePackages.econometrics</li>
    <li>octavePackages.financial</li>
    <li>octavePackages.fits</li>
    <li>octavePackages.fpl</li>
    <li>octavePackages.fuzzy-logic-toolkit</li>
    <li>octavePackages.ga</li>
    <li>octavePackages.general</li>
    <li>octavePackages.generate_html</li>
    <li>octavePackages.geometry</li>
    <li>octavePackages.gsl</li>
    <li>octavePackages.image</li>
    <li>octavePackages.instrument-control</li>
    <li>octavePackages.interval</li>
    <li>octavePackages.io</li>
    <li>octavePackages.linear-algebra</li>
    <li>octavePackages.lssa</li>
    <li>octavePackages.matgeom</li>
    <li>octavePackages.miscellaneous</li>
    <li>octavePackages.msh</li>
    <li>octavePackages.mvn</li>
    <li>octavePackages.nan</li>
    <li>octavePackages.ncarray</li>
    <li>octavePackages.netcdf</li>
    <li>octavePackages.nurbs</li>
    <li>octavePackages.ocl</li>
    <li>octavePackages.octclip</li>
    <li>octavePackages.octproj</li>
    <li>octavePackages.optics</li>
    <li>octavePackages.optim</li>
    <li>octavePackages.optiminterp</li>
    <li>octavePackages.quaternion</li>
    <li>octavePackages.queueing</li>
    <li>octavePackages.sockets</li>
    <li>octavePackages.splines</li>
    <li>octavePackages.statistics</li>
    <li>octavePackages.strings</li>
    <li>octavePackages.struct</li>
    <li>octavePackages.symbolic</li>
    <li>octavePackages.tsa</li>
    <li>octavePackages.video</li>
    <li>octavePackages.vrml</li>
    <li>octavePackages.windows</li>
    <li>octavePackages.zeromq</li>
    <li>octopus</li>
    <li>oobicpl</li>
    <li>openems</li>
    <li>openmm</li>
    <li>openmvg</li>
    <li>openturns</li>
    <li>ovito</li>
    <li>p2pool</li>
    <li>p4est</li>
    <li>p4est-dbg</li>
    <li>p4est-sc</li>
    <li>p4est-sc-dbg</li>
    <li>parmetis</li>
    <li>pmix</li>
    <li>precice</li>
    <li>pymol</li>
    <li>pymol.dist</li>
    <li>python310Packages.boltztrap2</li>
    <li>python310Packages.boltztrap2.dist</li>
    <li>python310Packages.cvxopt</li>
    <li>python310Packages.cvxopt.dist</li>
    <li>python310Packages.dask-mpi</li>
    <li>python310Packages.dask-mpi.dist</li>
    <li>python310Packages.eccodes</li>
    <li>python310Packages.fipy</li>
    <li>python310Packages.fipy.dist</li>
    <li>python310Packages.gpaw</li>
    <li>python310Packages.gpaw.dist</li>
    <li>python310Packages.h5netcdf</li>
    <li>python310Packages.h5netcdf.dist</li>
    <li>python310Packages.h5py-mpi</li>
    <li>python310Packages.h5py-mpi.dist</li>
    <li>python310Packages.hoomd-blue</li>
    <li>python310Packages.meep</li>
    <li>python310Packages.meshio</li>
    <li>python310Packages.meshio.dist</li>
    <li>python310Packages.mpi4py</li>
    <li>python310Packages.mpi4py.dist</li>
    <li>python310Packages.nest</li>
    <li>python310Packages.netcdf4</li>
    <li>python310Packages.netcdf4.dist</li>
    <li>python310Packages.openmm</li>
    <li>python310Packages.openturns</li>
    <li>python310Packages.osqp</li>
    <li>python310Packages.osqp.dist</li>
    <li>python310Packages.picos</li>
    <li>python310Packages.picos.dist</li>
    <li>python310Packages.python-csxcad</li>
    <li>python310Packages.python-csxcad.dist</li>
    <li>python310Packages.python-openems</li>
    <li>python310Packages.python-openems.dist</li>
    <li>python310Packages.scikit-survival</li>
    <li>python310Packages.scikit-survival.dist</li>
    <li>python310Packages.scikits-odes</li>
    <li>python310Packages.scikits-odes.dist</li>
    <li>python310Packages.sfepy</li>
    <li>python310Packages.sfepy.dist</li>
    <li>python310Packages.wrf-python</li>
    <li>python310Packages.wrf-python.dist</li>
    <li>python311Packages.boltztrap2</li>
    <li>python311Packages.boltztrap2.dist</li>
    <li>python311Packages.cartopy</li>
    <li>python311Packages.cartopy.dist</li>
    <li>python311Packages.cvxopt</li>
    <li>python311Packages.cvxopt.dist</li>
    <li>python311Packages.eccodes</li>
    <li>python311Packages.fiona</li>
    <li>python311Packages.fiona.dist</li>
    <li>python311Packages.folium</li>
    <li>python311Packages.folium.dist</li>
    <li>python311Packages.gdal</li>
    <li>python311Packages.geopandas</li>
    <li>python311Packages.geopandas.dist</li>
    <li>python311Packages.gpaw</li>
    <li>python311Packages.gpaw.dist</li>
    <li>python311Packages.h5netcdf</li>
    <li>python311Packages.h5netcdf.dist</li>
    <li>python311Packages.h5py-mpi</li>
    <li>python311Packages.h5py-mpi.dist</li>
    <li>python311Packages.meep</li>
    <li>python311Packages.meshio</li>
    <li>python311Packages.meshio.dist</li>
    <li>python311Packages.mpi4py</li>
    <li>python311Packages.mpi4py.dist</li>
    <li>python311Packages.nest</li>
    <li>python311Packages.netcdf4</li>
    <li>python311Packages.netcdf4.dist</li>
    <li>python311Packages.neuronpy</li>
    <li>python311Packages.openmm</li>
    <li>python311Packages.openturns</li>
    <li>python311Packages.osmnx</li>
    <li>python311Packages.osmnx.dist</li>
    <li>python311Packages.osqp</li>
    <li>python311Packages.osqp.dist</li>
    <li>python311Packages.picos</li>
    <li>python311Packages.picos.dist</li>
    <li>python311Packages.python-csxcad</li>
    <li>python311Packages.python-csxcad.dist</li>
    <li>python311Packages.python-openems</li>
    <li>python311Packages.python-openems.dist</li>
    <li>python311Packages.rasterio</li>
    <li>python311Packages.rasterio.dist</li>
    <li>python311Packages.scikit-survival</li>
    <li>python311Packages.scikit-survival.dist</li>
    <li>python311Packages.scikits-odes</li>
    <li>python311Packages.scikits-odes.dist</li>
    <li>python311Packages.wktutils</li>
    <li>python311Packages.wktutils.dist</li>
    <li>python311Packages.wrf-python</li>
    <li>python311Packages.wrf-python.dist</li>
    <li>quantum-espresso-mpi</li>
    <li>raxml-mpi</li>
    <li>rocalution</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>scalapack</li>
    <li>scotch</li>
    <li>siesta-mpi</li>
    <li>slurm</li>
    <li>slurm-spank-stunnel</li>
    <li>slurm-spank-x11</li>
    <li>slurm.dev</li>
    <li>suitesparse</li>
    <li>suitesparse.dev</li>
    <li>suitesparse.doc</li>
    <li>sundials</li>
    <li>sundials.examples</li>
    <li>toppler</li>
    <li>trafficserver</li>
    <li>trafficserver.man</li>
    <li>trilinos</li>
    <li>trilinos-mpi</li>
    <li>ucc</li>
    <li>ucx</li>
    <li>waifu2x-converter-cpp</li>
    <li>xfitter</li>
    <li>xmr-stak</li>
    <li>xmrig</li>
    <li>xmrig-mo</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
    <li>zfp</li>
  </ul>
</details>

SomeoneSerge committed:
config.cudaSupport: init option

SomeoneSerge committed:
treewide: consume config.cudaSupport as required

Eliminate uses of `config.cudaSupport or false` and alike, since the
option is now declared in config.nix with a default value

fd .nix -t f -x sed 's/config\.cudaSupport or false, cudaPackages [?] [{][}]/config.cudaSupport, cudaPackages ? { }/' '{}' -i
fd .nix -t f -x sed 's/config\.cudaSupport or false/config.cudaSupport/' '{}' -i
fd .nix -t f -x sed 's/cudaSupport = pkgs.config.cudaSupport/inherit (pkgs.config) cudaSupport/' '{}' -i
fd .nix -t f -x sed 's/cudaSupport = config.cudaSupport/inherit (config) cudaSupport/' '{}' -i

samuela commented:
> This PR does not introduce anything like config.cudaCapabilities which is left for future work

I'm not sure I understand... we already have `config.cudaCapabilities`, no? What would be the future work?

SomeoneSerge commented:
> I'm not sure I understand... we already have config.cudaCapabilities, no? What would be the future work?

An option definition in `pkgs/top-level/config.nix`

samuela reviewed (approved):
diff LGTM

SomeoneSerge commented:
Result of `nixpkgs-review pr 224068 --extra-nixpkgs-config '{ cudaCapabilities = [ "8.6" ]; }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

ConnorBaker commented:
Great work @SomeoneSerge! Merging.

SomeoneSerge commented:
Result of `nixpkgs-review pr 224068 --extra-nixpkgs-config '{ cudaCapabilities = [ "8.6" ]; cudaSupport = true; }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>85 packages marked as broken and skipped:</summary>
  <ul>
    <li>cntk</li>
    <li>ezminc</li>
    <li>gimpPlugins.exposureBlend</li>
    <li>gimpPlugins.resynthesizer</li>
    <li>octavePackages.fem-fenics</li>
    <li>octavePackages.image-acquisition</li>
    <li>octavePackages.level-set</li>
    <li>octavePackages.parallel</li>
    <li>octavePackages.sparsersb</li>
    <li>octavePackages.tisean</li>
    <li>octavePackages.vibes</li>
    <li>octavePackages.vrml</li>
    <li>python310Packages.cntk</li>
    <li>python310Packages.cntk.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.flax</li>
    <li>python310Packages.flax.dist</li>
    <li>python310Packages.optuna</li>
    <li>python310Packages.optuna.dist</li>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-aer</li>
    <li>python310Packages.qiskit-aer.dist</li>
    <li>python310Packages.qiskit-finance</li>
    <li>python310Packages.qiskit-finance.dist</li>
    <li>python310Packages.qiskit-ibmq-provider</li>
    <li>python310Packages.qiskit-ibmq-provider.dist</li>
    <li>python310Packages.qiskit-ignis</li>
    <li>python310Packages.qiskit-ignis.dist</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.qiskit-nature.dist</li>
    <li>python310Packages.qiskit-optimization</li>
    <li>python310Packages.qiskit-optimization.dist</li>
    <li>python310Packages.qiskit.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.scikit-tda</li>
    <li>python310Packages.scikit-tda.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python310Packages.vqgan-jax</li>
    <li>python310Packages.vqgan-jax.dist</li>
    <li>python310Packages.worldengine</li>
    <li>python310Packages.worldengine.dist</li>
    <li>python311Packages.cntk</li>
    <li>python311Packages.cntk.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.optuna</li>
    <li>python311Packages.optuna.dist</li>
    <li>python311Packages.qiskit-aer</li>
    <li>python311Packages.qiskit-aer.dist</li>
    <li>python311Packages.qiskit-finance</li>
    <li>python311Packages.qiskit-finance.dist</li>
    <li>python311Packages.qiskit-ibmq-provider</li>
    <li>python311Packages.qiskit-ibmq-provider.dist</li>
    <li>python311Packages.qiskit-ignis</li>
    <li>python311Packages.qiskit-ignis.dist</li>
    <li>python311Packages.qiskit-nature</li>
    <li>python311Packages.qiskit-nature.dist</li>
    <li>python311Packages.qiskit-optimization</li>
    <li>python311Packages.qiskit-optimization.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>python311Packages.worldengine</li>
    <li>python311Packages.worldengine.dist</li>
    <li>truecrack</li>
  </ul>
</details>
<details>
  <summary>160 packages failed to build:</summary>
  <ul>
    <li>dbcsr</li>
    <li>dl-poly-classic-mpi</li>
    <li>elpa</li>
    <li>frigate</li>
    <li>getdp</li>
    <li>gromacsDouble</li>
    <li>gromacsDoubleMpi</li>
    <li>openmvs</li>
    <li>parmetis</li>
    <li>perl534Packages.Tirex</li>
    <li>perl534Packages.Tirex.devdoc</li>
    <li>perl536Packages.Tirex</li>
    <li>perl536Packages.Tirex.devdoc</li>
    <li>petsc</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.blackjax</li>
    <li>python310Packages.blackjax.dist</li>
    <li>python310Packages.cvxpy</li>
    <li>python310Packages.cvxpy.dist</li>
    <li>python310Packages.gpt-2-simple</li>
    <li>python310Packages.gpt-2-simple.dist</li>
    <li>python310Packages.jaxopt</li>
    <li>python310Packages.jaxopt.dist</li>
    <li>python310Packages.opensfm</li>
    <li>python310Packages.opensfm.dist</li>
    <li>python310Packages.picos</li>
    <li>python310Packages.picos.dist</li>
    <li>python310Packages.pot</li>
    <li>python310Packages.pot.dist</li>
    <li>python310Packages.python-mapnik</li>
    <li>python310Packages.python-mapnik.dist</li>
    <li>python310Packages.qcodes</li>
    <li>python310Packages.qcodes-contrib-drivers</li>
    <li>python310Packages.qcodes-contrib-drivers.dist</li>
    <li>python310Packages.qcodes-loop</li>
    <li>python310Packages.qcodes-loop.dist</li>
    <li>python310Packages.qcodes.dist</li>
    <li>python310Packages.qutip</li>
    <li>python310Packages.qutip.dist</li>
    <li>python310Packages.sfepy</li>
    <li>python310Packages.sfepy.dist</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.skrl.dist</li>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.baselines</li>
    <li>python311Packages.baselines.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.cartopy</li>
    <li>python311Packages.cartopy.dist</li>
    <li>python311Packages.cleanlab</li>
    <li>python311Packages.cleanlab.dist</li>
    <li>python311Packages.ctranslate2</li>
    <li>python311Packages.ctranslate2.dist</li>
    <li>python311Packages.cvxpy</li>
    <li>python311Packages.cvxpy.dist</li>
    <li>python311Packages.django-bootstrap4</li>
    <li>python311Packages.django-bootstrap4.dist</li>
    <li>python311Packages.edward</li>
    <li>python311Packages.edward.dist</li>
    <li>python311Packages.fastai</li>
    <li>python311Packages.fastai.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.fiona</li>
    <li>python311Packages.fiona.dist</li>
    <li>python311Packages.folium</li>
    <li>python311Packages.folium.dist</li>
    <li>python311Packages.gdal</li>
    <li>python311Packages.geopandas</li>
    <li>python311Packages.geopandas.dist</li>
    <li>python311Packages.gpt-2-simple</li>
    <li>python311Packages.gpt-2-simple.dist</li>
    <li>python311Packages.hoomd-blue</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.mhcflurry</li>
    <li>python311Packages.mhcflurry.dist</li>
    <li>python311Packages.n3fit</li>
    <li>python311Packages.n3fit.dist</li>
    <li>python311Packages.numpyro</li>
    <li>python311Packages.numpyro.dist</li>
    <li>python311Packages.opensfm</li>
    <li>python311Packages.opensfm.dist</li>
    <li>python311Packages.osmnx</li>
    <li>python311Packages.osmnx.dist</li>
    <li>python311Packages.picos</li>
    <li>python311Packages.picos.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.pot</li>
    <li>python311Packages.pot.dist</li>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
    <li>python311Packages.qcodes</li>
    <li>python311Packages.qcodes-contrib-drivers</li>
    <li>python311Packages.qcodes-contrib-drivers.dist</li>
    <li>python311Packages.qcodes-loop</li>
    <li>python311Packages.qcodes-loop.dist</li>
    <li>python311Packages.qcodes.dist</li>
    <li>python311Packages.qutip</li>
    <li>python311Packages.qutip.dist</li>
    <li>python311Packages.rasterio</li>
    <li>python311Packages.rasterio.dist</li>
    <li>python311Packages.sfepy</li>
    <li>python311Packages.sfepy.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.spacy</li>
    <li>python311Packages.spacy-loggers</li>
    <li>python311Packages.spacy-loggers.dist</li>
    <li>python311Packages.spacy-lookups-data</li>
    <li>python311Packages.spacy-lookups-data.dist</li>
    <li>python311Packages.spacy-transformers</li>
    <li>python311Packages.spacy-transformers.dist</li>
    <li>python311Packages.spacy.dist</li>
    <li>python311Packages.tensorflow (python311Packages.tensorflow-build ,python311Packages.tensorflowWithCuda)</li>
    <li>python311Packages.tensorflow.dist (python311Packages.tensorflow-build.dist ,python311Packages.tensorflowWithCuda.dist)</li>
    <li>python311Packages.tensorflow-probability</li>
    <li>python311Packages.tensorflow-probability.dist</li>
    <li>python311Packages.tensorflowWithoutCuda</li>
    <li>python311Packages.tensorflowWithoutCuda.dist</li>
    <li>python311Packages.textacy</li>
    <li>python311Packages.textacy.dist</li>
    <li>python311Packages.textnets</li>
    <li>python311Packages.textnets.dist</li>
    <li>python311Packages.tf2onnx</li>
    <li>python311Packages.tf2onnx.dist</li>
    <li>python311Packages.tflearn</li>
    <li>python311Packages.tflearn.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python311Packages.trfl</li>
    <li>python311Packages.trfl.dist</li>
    <li>python311Packages.wandb</li>
    <li>python311Packages.wandb.dist</li>
    <li>python311Packages.wktutils</li>
    <li>python311Packages.wktutils.dist</li>
    <li>redpanda-server</li>
    <li>rtabmap</li>
    <li>saga</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>suitesparse_4_4</li>
    <li>sunshine</li>
    <li>t-rex</li>
    <li>xyce</li>
    <li>xyce.doc</li>
  </ul>
</details>
<details>
  <summary>430 packages built:</summary>
  <ul>
    <li>EBTKS</li>
    <li>apacheHttpdPackages.mod_tile</li>
    <li>apbs</li>
    <li>arrayfire</li>
    <li>bicgl</li>
    <li>bicpl</li>
    <li>cantor</li>
    <li>cdo</li>
    <li>ceres-solver</li>
    <li>ceres-solver.dev</li>
    <li>cholmod-extra</li>
    <li>cloudcompare</li>
    <li>colmap (colmapWithCuda)</li>
    <li>conglomerate</li>
    <li>cp2k</li>
    <li>eccodes</li>
    <li>elmerfem</li>
    <li>entwine</li>
    <li>febio</li>
    <li>fftwMpi</li>
    <li>fftwMpi.dev</li>
    <li>fftwMpi.info</li>
    <li>fftwMpi.man</li>
    <li>forge</li>
    <li>freecad</li>
    <li>g2o</li>
    <li>g2o.debug</li>
    <li>gama</li>
    <li>gdal (python310Packages.gdal)</li>
    <li>gegl</li>
    <li>gegl.dev</li>
    <li>gegl.devdoc</li>
    <li>gimp</li>
    <li>gimp-with-plugins</li>
    <li>gimp.dev</li>
    <li>gimpPlugins.bimp</li>
    <li>gimpPlugins.farbfeld</li>
    <li>gimpPlugins.fourier</li>
    <li>gimpPlugins.gap</li>
    <li>gimpPlugins.gimplensfun</li>
    <li>gimpPlugins.gmic</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.lqrPlugin</li>
    <li>gimpPlugins.texturize</li>
    <li>gimpPlugins.waveletSharpen</li>
    <li>globalarrays</li>
    <li>gmt</li>
    <li>gnome-photos</li>
    <li>gnome-photos.installedTests</li>
    <li>gnudatalanguage</li>
    <li>gplates</li>
    <li>gpt2tc</li>
    <li>grass</li>
    <li>gromacs</li>
    <li>gromacsMpi (gromacsCudaMpi)</li>
    <li>hal-hardware-analyzer</li>
    <li>hdf5-mpi</li>
    <li>hdf5-mpi.dev</li>
    <li>highfive-mpi</li>
    <li>hp2p</li>
    <li>hpcg</li>
    <li>hpl</li>
    <li>hpx</li>
    <li>hwloc</li>
    <li>hwloc.dev</li>
    <li>hwloc.doc</li>
    <li>hwloc.lib</li>
    <li>hwloc.man</li>
    <li>inormalize</li>
    <li>ior</li>
    <li>katago</li>
    <li>labplot</li>
    <li>lammps-mpi</li>
    <li>libLAS</li>
    <li>libminc</li>
    <li>libretranslate (python310Packages.libretranslate)</li>
    <li>libretranslate.dist (python310Packages.libretranslate.dist)</li>
    <li>librsb</li>
    <li>libtensorflow (libtensorflow.python)</li>
    <li>libvdwxc</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>migrate</li>
    <li>minc_tools</li>
    <li>minc_widgets</li>
    <li>mni_autoreg</li>
    <li>mpi</li>
    <li>mpich</li>
    <li>mprime</li>
    <li>mvapich</li>
    <li>mysql-workbench</li>
    <li>n3</li>
    <li>naev</li>
    <li>natron</li>
    <li>nco</li>
    <li>ncview</li>
    <li>nest-mpi</li>
    <li>netcdf</li>
    <li>netcdf-mpi</li>
    <li>netcdfcxx4</li>
    <li>netcdffortran</li>
    <li>neuron-full (python310Packages.neuronpy)</li>
    <li>neuron-mpi</li>
    <li>nwchem</li>
    <li>octave</li>
    <li>octaveFull</li>
    <li>octavePackages.arduino</li>
    <li>octavePackages.audio</li>
    <li>octavePackages.bim</li>
    <li>octavePackages.bsltl</li>
    <li>octavePackages.cgi</li>
    <li>octavePackages.communications</li>
    <li>octavePackages.control</li>
    <li>octavePackages.data-smoothing</li>
    <li>octavePackages.database</li>
    <li>octavePackages.dataframe</li>
    <li>octavePackages.dicom</li>
    <li>octavePackages.divand</li>
    <li>octavePackages.doctest</li>
    <li>octavePackages.econometrics</li>
    <li>octavePackages.financial</li>
    <li>octavePackages.fits</li>
    <li>octavePackages.fpl</li>
    <li>octavePackages.fuzzy-logic-toolkit</li>
    <li>octavePackages.ga</li>
    <li>octavePackages.general</li>
    <li>octavePackages.generate_html</li>
    <li>octavePackages.geometry</li>
    <li>octavePackages.gsl</li>
    <li>octavePackages.image</li>
    <li>octavePackages.instrument-control</li>
    <li>octavePackages.interval</li>
    <li>octavePackages.io</li>
    <li>octavePackages.linear-algebra</li>
    <li>octavePackages.lssa</li>
    <li>octavePackages.ltfat</li>
    <li>octavePackages.mapping</li>
    <li>octavePackages.matgeom</li>
    <li>octavePackages.miscellaneous</li>
    <li>octavePackages.msh</li>
    <li>octavePackages.mvn</li>
    <li>octavePackages.nan</li>
    <li>octavePackages.ncarray</li>
    <li>octavePackages.netcdf</li>
    <li>octavePackages.nurbs</li>
    <li>octavePackages.ocl</li>
    <li>octavePackages.octclip</li>
    <li>octavePackages.octproj</li>
    <li>octavePackages.optics</li>
    <li>octavePackages.optim</li>
    <li>octavePackages.optiminterp</li>
    <li>octavePackages.quaternion</li>
    <li>octavePackages.queueing</li>
    <li>octavePackages.signal</li>
    <li>octavePackages.sockets</li>
    <li>octavePackages.splines</li>
    <li>octavePackages.statistics</li>
    <li>octavePackages.stk</li>
    <li>octavePackages.strings</li>
    <li>octavePackages.struct</li>
    <li>octavePackages.symbolic</li>
    <li>octavePackages.tsa</li>
    <li>octavePackages.video</li>
    <li>octavePackages.windows</li>
    <li>octavePackages.zeromq</li>
    <li>octopus</li>
    <li>oobicpl</li>
    <li>openai-full</li>
    <li>openai-full.dist</li>
    <li>openems</li>
    <li>openmm</li>
    <li>openmvg</li>
    <li>openorienteering-mapper</li>
    <li>openturns</li>
    <li>ovito</li>
    <li>p2pool</li>
    <li>p4est</li>
    <li>p4est-dbg</li>
    <li>p4est-sc</li>
    <li>p4est-sc-dbg</li>
    <li>paraview</li>
    <li>pdal</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>pmix</li>
    <li>postgresql11JitPackages.postgis</li>
    <li>postgresql11JitPackages.postgis.doc</li>
    <li>postgresql11Packages.postgis</li>
    <li>postgresql11Packages.postgis.doc</li>
    <li>postgresql12JitPackages.postgis</li>
    <li>postgresql12JitPackages.postgis.doc</li>
    <li>postgresql12Packages.postgis</li>
    <li>postgresql12Packages.postgis.doc</li>
    <li>postgresql13JitPackages.postgis</li>
    <li>postgresql13JitPackages.postgis.doc</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql13Packages.postgis.doc</li>
    <li>postgresqlJitPackages.postgis (postgresql14JitPackages.postgis)</li>
    <li>postgresqlJitPackages.postgis.doc (postgresql14JitPackages.postgis.doc)</li>
    <li>postgresql14Packages.postgis</li>
    <li>postgresql14Packages.postgis.doc</li>
    <li>postgresql15JitPackages.postgis</li>
    <li>postgresql15JitPackages.postgis.doc</li>
    <li>postgresql15Packages.postgis</li>
    <li>postgresql15Packages.postgis.doc</li>
    <li>precice</li>
    <li>pymol</li>
    <li>pymol.dist</li>
    <li>python310Packages.argos-translate-files</li>
    <li>python310Packages.argos-translate-files.dist</li>
    <li>python310Packages.argostranslate</li>
    <li>python310Packages.argostranslate.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.baselines</li>
    <li>python310Packages.baselines.dist</li>
    <li>python310Packages.boltztrap2</li>
    <li>python310Packages.boltztrap2.dist</li>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.cartopy</li>
    <li>python310Packages.cartopy.dist</li>
    <li>python310Packages.cleanlab</li>
    <li>python310Packages.cleanlab.dist</li>
    <li>python310Packages.ctranslate2</li>
    <li>python310Packages.ctranslate2.dist</li>
    <li>python310Packages.cvxopt</li>
    <li>python310Packages.cvxopt.dist</li>
    <li>python310Packages.dask-mpi</li>
    <li>python310Packages.dask-mpi.dist</li>
    <li>python310Packages.datashader</li>
    <li>python310Packages.datashader.dist</li>
    <li>python310Packages.django-bootstrap4</li>
    <li>python310Packages.django-bootstrap4.dist</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.eccodes</li>
    <li>python310Packages.edward</li>
    <li>python310Packages.edward.dist</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.faster-whisper</li>
    <li>python310Packages.faster-whisper.dist</li>
    <li>python310Packages.fiona</li>
    <li>python310Packages.fiona.dist</li>
    <li>python310Packages.fipy</li>
    <li>python310Packages.fipy.dist</li>
    <li>python310Packages.folium</li>
    <li>python310Packages.folium.dist</li>
    <li>python310Packages.geopandas</li>
    <li>python310Packages.geopandas.dist</li>
    <li>python310Packages.gpaw</li>
    <li>python310Packages.gpaw.dist</li>
    <li>python310Packages.h5netcdf</li>
    <li>python310Packages.h5netcdf.dist</li>
    <li>python310Packages.h5py-mpi</li>
    <li>python310Packages.h5py-mpi.dist</li>
    <li>python310Packages.hoomd-blue</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.langchain</li>
    <li>python310Packages.langchain.dist</li>
    <li>python310Packages.meep</li>
    <li>python310Packages.meshio</li>
    <li>python310Packages.meshio.dist</li>
    <li>python310Packages.mhcflurry</li>
    <li>python310Packages.mhcflurry.dist</li>
    <li>python310Packages.mpi4py</li>
    <li>python310Packages.mpi4py.dist</li>
    <li>python310Packages.n3fit</li>
    <li>python310Packages.n3fit.dist</li>
    <li>python310Packages.nest</li>
    <li>python310Packages.netcdf4</li>
    <li>python310Packages.netcdf4.dist</li>
    <li>python310Packages.numpyro</li>
    <li>python310Packages.numpyro.dist</li>
    <li>python310Packages.openmm</li>
    <li>python310Packages.openturns</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.osmnx.dist</li>
    <li>python310Packages.osqp</li>
    <li>python310Packages.osqp.dist</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.plotnine.dist</li>
    <li>python310Packages.pygmt</li>
    <li>python310Packages.pygmt.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pyprecice</li>
    <li>python310Packages.pyprecice.dist</li>
    <li>python310Packages.pyslurm</li>
    <li>python310Packages.pyslurm.dist</li>
    <li>python310Packages.python-csxcad</li>
    <li>python310Packages.python-csxcad.dist</li>
    <li>python310Packages.python-openems</li>
    <li>python310Packages.python-openems.dist</li>
    <li>python310Packages.rasterio</li>
    <li>python310Packages.rasterio.dist</li>
    <li>python310Packages.scikit-survival</li>
    <li>python310Packages.scikit-survival.dist</li>
    <li>python310Packages.scikits-odes</li>
    <li>python310Packages.scikits-odes.dist</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-loggers</li>
    <li>python310Packages.spacy-loggers.dist</li>
    <li>python310Packages.spacy-lookups-data</li>
    <li>python310Packages.spacy-lookups-data.dist</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.spacy-transformers.dist</li>
    <li>python310Packages.spacy.dist</li>
    <li>python310Packages.tensorflow (python310Packages.tensorflow-build ,python310Packages.tensorflowWithCuda)</li>
    <li>python310Packages.tensorflow.dist (python310Packages.tensorflow-build.dist ,python310Packages.tensorflowWithCuda.dist)</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python310Packages.tensorflow-probability</li>
    <li>python310Packages.tensorflow-probability.dist</li>
    <li>python310Packages.tensorflowWithoutCuda</li>
    <li>python310Packages.tensorflowWithoutCuda.dist</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.textacy.dist</li>
    <li>python310Packages.textnets</li>
    <li>python310Packages.textnets.dist</li>
    <li>python310Packages.tf2onnx</li>
    <li>python310Packages.tf2onnx.dist</li>
    <li>python310Packages.tflearn</li>
    <li>python310Packages.tflearn.dist</li>
    <li>python310Packages.translatehtml</li>
    <li>python310Packages.translatehtml.dist</li>
    <li>python310Packages.trfl</li>
    <li>python310Packages.trfl.dist</li>
    <li>python310Packages.umap-learn</li>
    <li>python310Packages.umap-learn.dist</li>
    <li>python310Packages.wandb</li>
    <li>python310Packages.wandb.dist</li>
    <li>python310Packages.wktutils</li>
    <li>python310Packages.wktutils.dist</li>
    <li>python310Packages.wrf-python</li>
    <li>python310Packages.wrf-python.dist</li>
    <li>python311Packages.boltztrap2</li>
    <li>python311Packages.boltztrap2.dist</li>
    <li>python311Packages.cvxopt</li>
    <li>python311Packages.cvxopt.dist</li>
    <li>python311Packages.dask-mpi</li>
    <li>python311Packages.dask-mpi.dist</li>
    <li>python311Packages.eccodes</li>
    <li>python311Packages.fipy</li>
    <li>python311Packages.fipy.dist</li>
    <li>python311Packages.gpaw</li>
    <li>python311Packages.gpaw.dist</li>
    <li>python311Packages.h5netcdf</li>
    <li>python311Packages.h5netcdf.dist</li>
    <li>python311Packages.h5py-mpi</li>
    <li>python311Packages.h5py-mpi.dist</li>
    <li>python311Packages.meep</li>
    <li>python311Packages.meshio</li>
    <li>python311Packages.meshio.dist</li>
    <li>python311Packages.mpi4py</li>
    <li>python311Packages.mpi4py.dist</li>
    <li>python311Packages.nest</li>
    <li>python311Packages.netcdf4</li>
    <li>python311Packages.netcdf4.dist</li>
    <li>python311Packages.neuronpy</li>
    <li>python311Packages.openmm</li>
    <li>python311Packages.openturns</li>
    <li>python311Packages.osqp</li>
    <li>python311Packages.osqp.dist</li>
    <li>python311Packages.pygmt</li>
    <li>python311Packages.pygmt.dist</li>
    <li>python311Packages.pyprecice</li>
    <li>python311Packages.pyprecice.dist</li>
    <li>python311Packages.pyslurm</li>
    <li>python311Packages.pyslurm.dist</li>
    <li>python311Packages.python-csxcad</li>
    <li>python311Packages.python-csxcad.dist</li>
    <li>python311Packages.python-openems</li>
    <li>python311Packages.python-openems.dist</li>
    <li>python311Packages.scikit-survival</li>
    <li>python311Packages.scikit-survival.dist</li>
    <li>python311Packages.scikits-odes</li>
    <li>python311Packages.scikits-odes.dist</li>
    <li>python311Packages.wrf-python</li>
    <li>python311Packages.wrf-python.dist</li>
    <li>pytrainer</li>
    <li>pytrainer.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>qmapshack</li>
    <li>quantum-espresso-mpi</li>
    <li>raxml-mpi</li>
    <li>rocalution</li>
    <li>scalapack</li>
    <li>scotch</li>
    <li>siesta-mpi</li>
    <li>slurm</li>
    <li>slurm-spank-stunnel</li>
    <li>slurm-spank-x11</li>
    <li>slurm.dev</li>
    <li>suitesparse</li>
    <li>suitesparse.dev</li>
    <li>suitesparse.doc</li>
    <li>sumo</li>
    <li>sundials</li>
    <li>sundials.examples</li>
    <li>toppler</li>
    <li>trafficserver</li>
    <li>trafficserver.man</li>
    <li>trilinos</li>
    <li>trilinos-mpi</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>ucc</li>
    <li>ucx</li>
    <li>udig</li>
    <li>vpv</li>
    <li>waifu2x-converter-cpp</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
    <li>xfitter</li>
    <li>xmr-stak</li>
    <li>xmrig</li>
    <li>xmrig-mo</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
    <li>zfp</li>
  </ul>
</details>
